client/src/components/v2/CameraRecording.tsx

```diff
@@ -0,0 +1,98 @@
+import { VideoFeedStatus } from 'api/v2/camera';
+import React from 'react';
+import { Badge, Button, Card, Col, Row } from 'react-bootstrap';
+import styles from './CameraRecording.module.css';
+import CameraRecordingStatus, { CameraRecordingStatusProps } from './CameraRecordingStatus';
+
+export type CameraRecordingPropT = VideoFeedStatus & CameraRecordingStatusProps & { ip: string };
+
+export interface CameraRecordingProps {
+  /** Information for primary device */
+  primary: CameraRecordingPropT
+  /** Information for secondary device */
+  secondary: CameraRecordingPropT
+  startRecording: () => void
+  stopRecording: () => void
+}
+
+/**
+ * Camera recording buttons for starting and stopping video recording.
+ *
+ * Starts/stops recording for both displays.
+ *
+ * This is a feature intended for V3 but is currently in V2 for testing.
+ *
+ * @param props Props
+ * @returns Component
+ */
+export default function CameraRecording({
+  primary,
+  secondary,
+  startRecording,
+  stopRecording
+}: CameraRecordingProps): JSX.Element {
+  // Check if at least one camera's video feed is online
+  const canRecord = primary.online || secondary.online;
+
+  return (
+    <Card>
+      <Card.Body>
+        <Card.Title>Recording Controls</Card.Title>
+        <Row className={styles.row}>
+          {/* Device recording controls UI for primary device */}
+          <Col className={styles.col} key='primary' sm={6}>
+            <Card.Subtitle className={styles.subtitle}>
+              {/* Name */}
+              <span className={styles.deviceName}>
+                Primary
+                </span>
+              {/* Online status */}
+              <Badge className={styles.indicator} pill variant={primary.online ? 'success' : 'danger'}>
+                {primary.online ? 'Online' : 'Offline'}
+              </Badge>
+              {/* IP */}
+              <div className={styles.push}>{primary.ip}</div>
+            </Card.Subtitle>
+            {/* Fields */}
+            <CameraRecordingStatus statusFormatted={primary?.statusFormatted} />
+          </Col>
+          {/* Device recording controls UI for secondary device */}
+          <Col className={styles.col} key='secondary' sm={6}>
```

Comment:
Reckon you could cut down this repeated code? Maybe put the `Col` inside a `['pirmary', 'secondary'].forEach(...)`

Reply:
done but i'm not sure if this is that much better than what we had before, it does reduce the repeated code but increases the complexity.
e805a93

Reply:
Hmm I didn't realise we'd have to do the `device`/`deviceName` thing, but I reckon it's still worthwhile! Makes it clear that the two columns should behave identically.